### ISSUE TITLE
Improve the column right-click menu and add an Edit Cell(s) option

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#8239]
@lloyddewit and @N-thony I'd like your confirmation that this is a sensible idea?

![image](https://user-images.githubusercontent.com/11226469/227454278-6164e223-3b89-4539-a82f-c56bb95b53f7.png)

This is a data frame with all the months where the whole month was missing.  I am quite pleased with how easily it was able to get this.  I would now like staff to investigate the situation, by station and month by month to see if they can locate the missing data.  So I'd like a comment on what they find each station/year/month in the last column.

Now you can currently type into each cell in turn already, but I am wondering if it could be useful to also have an Edit Cell(s) option, below the Delete Cell(s) option.  This would sort of be similar to Excel, where you have the entry are the top of a sheetm where you can type and that goes automatically into the current cell.

This would usually be for a single cell, but ut could, like the delete cell(s) also put the same text into a block of cells.

It would open s imple dialogue with the cell selected and the current contents - for a single cell.  For a block of cells it would just give the column and rows.  Then (like the comment dialogue) you can type away.

By the way, when editing this right-click menu could we take the opportunity to delete the Paste option.  Paste seems never to be finisherd, and (even when it is) I don't want it there.  I don't want to particularly encourage pasting and the option is fine in the Edit menu.

**Comment by lloyddewit** [comment]
@rdstern If I was a user, I think I would intuitively enter a value into one cell and then try and paste the value into multiple cells (this is what I would do in a spreadsheet). Would it be better to make the data grid writable and get paste working correctly?
I also sometimes use the feature in Excel where you specify a range in the top-left corner.

However if you have reasons to dislike paste in the data grid, then I have no objections to the proposals above. I am just concerned that its different to how spreadsheets behave.

**Comment by rdstern** [comment]
@derekagorhom could you please try this one?
In the cells right-click menu effectively move the paste option to just below the Delete Cell(s) and change it into Edit Cell(s).
Then make a very simple dialogue, that takes the content of the current cell and allows it to be changed.  If multiple cells have been selected, then it takes the content of the first cell.  (It is mainly designed for a single cell.)  Then it replaces the content of the chosen cell by what is in the field.  It is sort of like the add comment, but much simpler, so doesn't need all the controls of an ordinary dialogue.

**Comment by rdstern** [comment]
@derekagorhom, once you are back from leave, I have changed to title of this issue, because the change in the right-click menu is now a first priority.  When preparing an introductory tutorial for beginners it became clear that we are complicating user's lives that the right-click menu is changed dpeneding on whether the focus is in the top name field, or below.

So the right-click menu should always be the same and as follows:

![image](https://user-images.githubusercontent.com/11226469/232976276-cdff4af7-22cc-4fc3-8bf1-f966d0b19335.png)

The top 4 items remain as shown here.

Then there is a horizontal line.  Then
Edit Cell(s)
Delete Cell(s)
Then the Paste is deleted.  So the horizontal line and the Edit and Delete Cell(s) are simply instead of Paste!  They are just above the existing horizontal line.  And then, as shown above, there is Convert to Factor.

Just one more thing.  When you are at the top row, as shown above, then the Edit Cell(s) and Delete Cell(s) entries are disabled.  Neat eh?

You may need help, so please check (next week) with @lloyddewit .  He is also on leave this week.

**Comment by rdstern** [comment]
@derekagorhom this issue covers both your current work on Edit Cell and the eight-click menu.

Note to make progress I have changed your Edit Cell(s) into Edit Cell.  So (at least for now) just do it for a single cell.  Then we will merge that one and you can concentrate on the menu.

Then - as Stephen mentions above, we could complete Patrick's Paste options - pull request from ages ago - and use that the Copy a cell and paste into multiple cells.

**Comment by rdstern** [comment]
@derekagorhom here are the new aspects, after the Edit Cell has been merged.
a) In the `factor `add a last option NA, to be able to include a missing value when editing a factor.
b) Add a special facility for a `logical `variable - like a Factor, but just allowing FAL:SE, or TRUE, or NA.
c) If a `Date` variable include the special control for a date as we discussed.  There is the date control in the Climatic > Tidt > Enter/Edit data
d) (Possibly for later - is there a similar time and a date-time control.  Only if quick.  
e) Consider extending to a marked set of cells if possible (easy?) as follows.  If so the editing is controlled by the edit to the top cell, or the top left cell if a rectangle.  Then, if a single column the new value is copied into each further cell.  If a rectangle, then either don't allow - so just one variable.  If allow then only if all variables are the same type.  If so, then can copy as toy do delete for a rectangle.  If not, then either give an error, or just implement for the one Variable.
